Decsys/Data/Entities/BaseComponent.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Decsys.Data.Entities
+{
+    public abstract class BaseComponent
+    {
+
+
+        /// <summary>
+        /// Create a Component of the specified type.
+        /// </summary>
+        /// <param name="type">The component type.</param>
+        /// Constructor with paramters should be in base class but gives error? 
+        /*public BaseComponent(string type)
```

Comment:
this constructor should be used because it sets a property of the base class, so ensures consistency in derived classes